commodore/dependency_mgmt.py

```diff
@@ -65,12 +64,38 @@ def _discover_components(cfg, inventory_path):
     return sorted(components)
 
 
-def _fetch_component(cfg, component):
-    repository_url = f"{cfg.global_git_base}/commodore-components/{component}.git"
-    target_directory = P('dependencies') / component
-    repo = git.clone_repository(repository_url, target_directory)
-    cfg.register_component(component, repo)
-    create_component_symlinks(component)
+def _read_component_urls(cfg, component_names):
+    components = []
+    component_urls = {}
+    if cfg.debug:
+        click.echo(f"   > Read commodore config file {cfg.config_file}")
+    try:
+        commodore_config = yaml_load(cfg.config_file)
+    except Exception as e:
+        raise click.ClickException(f"Could not read Commodore configuration: {e}") from e
+    if commodore_config is None:
+        click.secho('Empty Commodore config file', fg='yellow')
+    else:
+        for component_override in commodore_config.get('components', []):
+            if cfg.debug:
+                click.echo(f"   > Found override for component {component_override['name']}:")
+                click.echo(f"     Using URL {component_override['url']}")
+            component_urls[component_override['name']] = component_override['url']
+    for component_name in component_names:
+        repository_url = \
+            component_urls.get(
+                component_name,
+                f"{cfg.default_component_base}/{component_name}.git")
+        target_directory = P('dependencies') / component_name
+        component = Component(
+            name=component_name,
+            repo=None,
+            version='master',
+            repo_url=repository_url,
+            target_directory=target_directory,
+        )
+        components.append(component)
```

Comment:
*This comment is informational only, no changes required*

I did a short experiment with calling `cfg.register_component` here, and changing `fetch_components` to use `cfg.get_components()`, and an additional method `cfg.set_component_repo()`. However, the end result of that change does not really improve readability of the code, so I'm fine with leaving it like this for now.